Sources/Foundation/NSDictionary.swift

```diff
@@ -372,46 +372,47 @@ open class NSDictionary : NSObject, NSCopying, NSMutableCopying, NSSecureCoding,
+            var line = nextLevelIndentation
+
+            switch key {
+            case let nsArray as NSArray:
+                line += nsArray.description(withLocale: locale, indent: level + 1)
+            case let nsDate as Date:
+                line += nsDate.description(with: locale)
+            case let nsDecimalNumber as NSDecimalNumber:
+                line += nsDecimalNumber.description(withLocale: locale)
+            case let nsDictionary as NSDictionary:
+                line += nsDictionary.description(withLocale: locale, indent: level + 1)
+            case let nsOderedSet as NSOrderedSet:
+                line += nsOderedSet.description(withLocale: locale, indent: level + 1)
+            case let nsSet as NSSet:
+                line += nsSet.description(withLocale: locale)
+            default:
                 line += "\(key)"
             }
 
             line += " = "
 
             let object = self.object(forKey: key)!
-            if object is NSArray {
-                line += (object as! NSArray).description(withLocale: locale, indent: level + 1)
-            } else if object is Date {
-                line += (object as! NSDate).description(with: locale)
-            } else if object is NSDecimalNumber {
-                line += (object as! NSDecimalNumber).description(withLocale: locale)
-            } else if object is NSDictionary {
-                line += (object as! NSDictionary).description(withLocale: locale, indent: level + 1)
-            } else if object is NSOrderedSet {
-                line += (object as! NSOrderedSet).description(withLocale: locale, indent: level + 1)
-            } else if object is NSSet {
-                line += (object as! NSSet).description(withLocale: locale)
-            } else {
-                if let hashableObject = object as? Dictionary<AnyHashable, Any> {
-                    line += hashableObject._nsObject.description(withLocale: nil, indent: level+1)
-                } else {
-                    line += "\(object)"
-                }
+            switch object {
+            case let nsArray as NSArray:
+                line += nsArray.description(withLocale: locale, indent: level + 1)
+            case let nsDate as NSDate:
+                line += nsDate.description(with: locale)
+            case let nsDecimalNumber as NSDecimalNumber:
+                line += nsDecimalNumber.description(withLocale: locale)
+            case let nsDictionary as NSDictionary:
+                line += nsDictionary.description(withLocale: locale, indent: level + 1)
+            case let nsOrderedSet as NSOrderedSet:
+                line += nsOrderedSet.description(withLocale: locale, indent: level + 1)
+            case let nsSet as NSSet:
+                line += nsSet.description(withLocale: locale)
+            case let hashableObject as Dictionary<AnyHashable, Any>:
+                line += hashableObject._nsObject.description(withLocale: nil, indent: level+1)
```

Comment:
Minor nit, fix the white space around `level+1`, thanks.

Reply:
fixed